mapper/orm/src/main/java/org/hibernate/search/mapper/orm/event/impl/HibernateSearchEventListener.java

```diff
@@ -145,6 +148,15 @@ public void onFlush(FlushEvent event) {
 //		}
 	}
 
+	@Override
+	public void onClear(ClearEvent event) {
+		EventSource session = event.getSession();
+		// skip the clearNotPrepared operation in case of rollback
```

Comment:
Note for later: this is a piece of code that should be changed in another PR related to adding support for events outside of transactions.